step.go

```diff
@@ -1,11 +1,75 @@
 package allure
 
-type Step struct {
-	Name          string
-	Status        string
-	StatusDetails StatusDetails
-	Stage         string
-	Start         int64
-	Stop          int64
-	Attachements  []Attachment
+import (
+	"fmt"
+	"github.com/jtolds/gls"
+	"log"
+	"time"
+)
+
+type stepObject struct {
+	Name          string       `json:"name,omitempty"`
+	Status        string       `json:"status,omitempty"`
+	Stage         string       `json:"stage"`
+	ChildrenSteps []stepObject `json:"steps"`
+	Attachements  []Attachment `json:"attachments"`
+	Parameters    []string     `json:"parameters"`
+	Start         int64        `json:"start"`
+	Stop          int64        `json:"stop"`
+}
+
+//"status": "passed",
+//      "stage": "finished",
+//      "steps": [
+//        {
+//          "name": "Other stuff",
+//          "status": "passed",
+//          "stage": "finished",
+//          "steps": [],
+//          "attachments": [],
+//          "parameters": [],
+//          "start": 1572368980980,
+//          "stop": 1572368980981
+//        }
+//      ],
+//      "attachments": [],
+//      "parameters": [],
+//      "start": 1572368980979,
+//      "stop": 1572368980982
+
+func (s *stepObject) GetSteps() []stepObject {
+	return s.ChildrenSteps
+}
+
+func (s *stepObject) AddStep(step stepObject) {
+	s.ChildrenSteps = append(s.ChildrenSteps, step)
+}
+
+func Step(description string, action func()) {
+	log.Println(description)
+	step := newStep()
+	step.Name = description
+	step.Start = time.Now().Unix()
+	defer func() {
+		step.Stop = time.Now().Unix()
```

Comment:
I guess this needs a value in Ms. Please replace with getTimestampMs()

Reply:
Done.